Sources/Common/DataModel/DataSetAttributes/FieldData.js

```diff
@@ -79,18 +80,46 @@ function vtkFieldData(publicAPI, model) {
   publicAPI.getCopyFieldFlags = () => model.copyFieldFlags;
   publicAPI.getFlag = (arrayName) => model.copyFieldFlags[arrayName];
   publicAPI.passData = (other, fromId = -1, toId = -1) => {
-    other.getArrays().forEach((arr, idx) => {
+    other.getArrays().forEach((arr) => {
       const copyFlag = publicAPI.getFlag(arr.getName());
       if (
         copyFlag !== false &&
         !(model.doCopyAllOff && copyFlag !== true) &&
         arr
       ) {
-        const destArr = publicAPI.getArrayByName(arr.getName());
-        if ((fromId < 1 && toId < 1) || !destArr) {
-          publicAPI.addArray(arr);
-        } else if (idx >= fromId && (toId > -1 || idx < toId)) {
-          destArr.setTuple(idx, arr.getTuple(idx));
+        let destArr = publicAPI.getArrayByName(arr.getName());
+        if (!destArr) {
+          if (fromId < 0 || fromId > arr.getNumberOfTuples()) {
+            publicAPI.addArray(arr);
+          } else {
+            const ncomps = arr.getNumberOfComponents();
+            let newSize = arr.getNumberOfValues();
+            const tId = toId > -1 ? toId : fromId;
+            if (newSize < tId * ncomps) {
+              newSize = (tId + 1) * ncomps;
+            }
+            destArr = vtkDataArray.newInstance({
+              name: arr.getName(),
+              dataType: arr.getDataType(),
+              numberOfComponents: arr.getNumberOfComponents(),
+              size: newSize,
+            });
+            destArr.setTuple(tId, arr.getTuple(fromId));
```

Comment:
I'm not sure I agree here... Can you explain? Is that array supposed to only have 1 tuple?

I would use API directly on the typed array.

Reply:
I guess if it is only for 1 tuple, then it is fine. 
But if it was to copy on section of an array to another. The typed array would have everything.